troposphere/static/js/components/modals/project/ProjectAddImageModal.react.js

```diff
@@ -85,20 +104,73 @@ define(function (require) {
           <p> {this.props.image.get('name')} </p>
         );
       },
+      renderExistingProjects: function () {
+          if(this.state.existing == null) {
+              return (<div className='loading' />);
+          }
+          project_divs = this.state.existing.map(function(existing_project) {
+              return (
+                <div id={existing_project.cid}>
+                  {existing_project.get('project').name}
+                </div>
+              );
+          });
+
+          return (
+             <div className='form-group'>
+               <label htmlFor='existing-project'>Added to Project(s)</label>
+               <div id='existing-project'>
+                 {project_divs}
+                </div>
+             </div>
+          );
+      },
+      filterRemainingProjects: function() {
+          if(this.state.projects == null || this.state.existing == null) {
+              return [];
+          }
+          var self = this;
+          var project_arr = this.state.projects.filter(function(project) {
+              needle = project.id;
+              haystack_matches = self.state.existing.filter(
+                  function(existing_project) {
+                      project_id = existing_project.get('project').id;
+                      return project_id == needle;
+              });
+              return haystack_matches.length == 0;
+          });
+          projects = new Backbone.Collection(project_arr);
+          return projects;
+      },
       renderProjects: function () {
-          if(this.state.projects == null) {
-              projects = stores.ProjectStore.getAll();
+          if(this.state.projects == null || this.state.existing == null) {
```

Comment:
In a "Trust No One" view, you might say `if (!this.state.projects || !this.state.existing)` to catch a potential situation where either might be `undefined`. But here, with `getInitialState` defining both - we only need concern ourselves with being equal to `null`. 

This is mainly me thinking out loud 👂  please ignore.